src/core/src/preprocess/preprocess_impls.cpp

```diff
@@ -369,56 +368,36 @@ void OutputInfo::OutputInfoImpl::build(ov::ResultVector& results) {
         node = std::get<0>(action_result);
         post_processing_applied = true;
     }
-    // Restore tensor names
-    node.get_tensor().set_names(start_out_node_names);
+
     auto orig_parent = result->get_input_source_output(0).get_node_shared_ptr();
-    bool reset_orig_friendly_name = false;
+    // Move result tensor names from previous input to new
+    const auto result_input_names = result->get_input_tensor(0).get_names();
+    result->get_input_tensor(0).set_names({});
+    node.get_tensor().set_names(result_input_names);
+
     if (!post_processing_applied) {
         return;
     }
+
     if (orig_parent->get_output_size() == 1) {
         node.get_node_shared_ptr()->set_friendly_name(orig_parent->get_friendly_name());
-        reset_orig_friendly_name = true;
+
+        // Reset friendly name of input node to avoid names collision
+        // when there is at a new node inserted by post-processing steps
+        // If no new nodes are inserted by post-processing, then we need to preserve friendly name of input
+        // as it's required for old API correct work
```

Comment:
For PPP compatibility mode it has to be kept (I don't want to change this functionality in this PR).

For IRv11 it can be considered as  a bug which is solved in [PR #23713](https://github.com/openvinotoolkit/openvino/pull/23713/files#diff-7c4d81a6d4bb639ff9af23a3a8f09cda9ec474f9fa1d6acf988c21b9c5ede74dR373). 

The Node's (layer) name should not be moved which was reported in [CVS-127482](https://jira.devtools.intel.com/browse/CVS-127482)